`nocover/test_respects_max_examples_in_database_usage` is flaky

```python-traceback
_________________ test_respects_max_examples_in_database_usage _________________
[gw0] linux2 -- Python 2.7.13 /home/vsts/work/1/s/hypothesis-python/.tox/pypy-full/bin/python
Traceback (most recent call last):
  File "/home/vsts/work/1/s/hypothesis-python/tests/nocover/test_database_usage.py", line 124, in test_respects_max_examples_in_database_usage
    assert len(all_values(database)) > 10
AssertionError: assert 10 > 10
 +  where 10 = len(set([compatbytes(...), ...]))
 +    where set([compatbytes(...), ...]) = all_values(InMemoryExampleDatabase({'a database key': set([compatbytes(...)])}))
```

[Via this CI failure](https://dev.azure.com/HypothesisWorks/Hypothesis/_build/results?buildId=3107&view=logs&jobId=d1cafe0c-32b1-5c9c-e54f-8f4972aa4ad6&taskId=32d1317c-102d-5d27-f947-f1b7daa52934&lineStart=485&lineEnd=493&colStart=1&colEnd=1).  I don't *think* this is a Py2-only or pypy-only issue, or both, but it might be.  [This one](https://dev.azure.com/HypothesisWorks/Hypothesis/_build/results?buildId=3097&view=logs&jobId=d1cafe0c-32b1-5c9c-e54f-8f4972aa4ad6&taskId=32d1317c-102d-5d27-f947-f1b7daa52934&lineStart=485&lineEnd=493&colStart=1&colEnd=1) also happened on pypy2, so...

--- a/hypothesis-python/tests/nocover/test_database_usage.py
+++ b/hypothesis-python/tests/nocover/test_database_usage.py
@@ -24,6 +24,7 @@ from hypothesis import assume, core, find, given, settings
 from hypothesis.database import ExampleDatabase, InMemoryExampleDatabase
 from hypothesis.errors import NoSuchExample, Unsatisfiable
 from hypothesis.internal.compat import hbytes
+from hypothesis.internal.entropy import deterministic_PRNG
 from tests.common.utils import all_values, non_covering_examples
 
 
@@ -120,7 +121,8 @@ def test_respects_max_examples_in_database_usage():
         except NoSuchExample:
             pass
 
-    stuff()
+    with deterministic_PRNG():
+        stuff()
     assert len(all_values(database)) > 10
     do_we_care = False
     counter[0] = 0

